src/libstd/sys/windows/mod.rs

```diff
@@ -72,10 +72,13 @@ pub fn decode_error_kind(errno: i32) -> ErrorKind {
     }
 }
 
-fn to_utf16_os(s: &OsStr) -> Vec<u16> {
-    let mut v: Vec<_> = s.encode_wide().collect();
-    v.push(0);
-    v
+pub fn to_u16s(s: &OsStr) -> Option<Vec<u16>> {
```

Comment:
It's fine to take `T: AsRef<OsStr>` and I don't think the error message here is necessarily critical enough to tailor to each use case